src/cache/surface_albedo.jl

```diff
@@ -0,0 +1,195 @@
+# we are ignoring the volume reflectance for now
+abstract type SurfaceAlbedoModel end
+
+"""
+    Base.@kwdef struct ConstantAlbedo{FT} <: SurfaceAlbedoModel
+
+A constant surface albedo model. The default value is 0.38. It is used purely for idealized experiments.
+"""
+Base.@kwdef struct ConstantAlbedo{FT} <: SurfaceAlbedoModel
+    α::FT = 0.38
+end
+
+"""
+    Base.@kwdef struct RegressionFunctionAlbedo{FT} <: SurfaceAlbedoModel
+
+A regression function-based surface albedo model of Jin et al. (2011, referred to as J11).
+
+
+"""
+Base.@kwdef struct RegressionFunctionAlbedo{FT} <: SurfaceAlbedoModel
+    n::FT = 1.34                                                  # relative refractive index of water and air (n = n_w/n_a) TODO: f(wavelength) for a spectrally dependent scheme
+    n0::FT = 1.34                                                 # refractive index of water for visible light
+    p::Vector{FT} = [
+        0.0152,
+        -1.7873,
+        6.8972,
+        -8.5778,
+        4.071,
+        -7.6446,
+        0.1643,
+        -7.8409,
+        -3.5639,
+        -2.3588,
+        10.0538,
+    ]                                                             # regression coefficients of J11 for eq. 4
+    wave_slope::Function = u -> sqrt(FT(0.003) + FT(0.00512) * u) # mean wave slope distribution width (Cox-Munk model, J11 Eq. 2)
```

Comment:
This will also not go on a GPU, you need to make it a parameterized type `F <: Function`